emu/src/cpu/arm7tdmi.rs

```diff
@@ -2141,4 +2196,100 @@ mod tests {
             assert_eq!(cpu.memory.lock().unwrap().read_half_word(102), 0xFF);
         }
     }
+
+    #[test]
+    fn check_load_store_sign_extend_byte_halfword() {
+        {
+            let mut cpu = Arm7tdmi::default();
+            let op_code: u16 = 0b0101_0_0_1_000_001_010;
+            let op_code: ThumbModeOpcode = cpu.decode(op_code);
+            assert_eq!(
+                op_code.instruction,
+                ThumbModeInstruction::LoadStoreSignExtByteHalfword {
+                    h: false,
+                    s: false,
+                    ro: 0,
+                    rb: 1,
+                    rd: 2
+                }
+            );
+
+            cpu.registers.set_register_at(0, 10);
+            cpu.registers.set_register_at(1, 100);
+            cpu.registers.set_register_at(2, 0xF000_F0FF);
+
+            cpu.execute_thumb(op_code);
```

Comment:
We can split test for decoding from execution

Reply:
As discussed I did a first try with table tests, let's see